### PR TITLE
feat: make report composition async to the harbor request

### PR DESCRIPTION
**Pull request by bradleyjones**
Fixes: #50

### CODE REVIEW

File: pkg/adapter/anchore/result_store_test.go [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+		{
+			name:   "result found and scan complete",
+			fields: fields{Results: map[string]VulnerabilityResult{"test1": {ScanId: "test1", IsComplete: true}}},
+			args:   args{"test1"},
+			want:   true,
+		},
+		{
+			name:   "result found and scan incomplete",
+			fields: fields{Results: map[string]VulnerabilityResult{"test1": {ScanId: "test1", IsComplete: false}}},
+			args:   args{"tes"},
```
Comment (bot190) [inline pkg/adapter/anchore/result_store_test.go:34]: Is this correct?

**Review by bot190 — COMMENTED** [review]
This approach looks good to me!

File: pkg/adapter/anchore/result_store_test.go (hunk 1 above)
Comment (bradleyjones) [reply pkg/adapter/anchore/result_store_test.go:34]: it is not, good catch

File: pkg/adapter/anchore/result_store.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	return newStore
+}
+
+func (m MemoryResultStore) HasResult(scanId string) bool {
+	found, ok := m.Results[scanId]
+	log.Debugf("HasResult: %v", found)
+	return ok && found.IsComplete
+}
+
+func (m MemoryResultStore) GetResult(scanId string) (VulnerabilityResult, bool) {
```
[the remark this answers is not in the document]
Comment (bradleyjones) [reply pkg/adapter/anchore/result_store.go:54]: Good call, will make the change quickly

**Review by zhill — APPROVED** [review]
lgtm! There are a few tweaks to debug lines etc I'd like to make, but let's handle those after the merge. Just some debugging output things to make it easier to see what's going on in the flow at both info and debug level.

**Pull request merged**